### PR TITLE
SNOW-1620436 Improved to_datetime to handle all local input cases and improved time series notebook tests

### PR DESCRIPTION
**Pull request by sfc-gh-azhan**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1620436

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

This pull request includes several changes to improve the handling of datetime conversions and related tests in the Snowpark Modin integration. The most important changes include enhancements to the `to_datetime` function, updates to internal utility messages, and adjustments to test cases for better accuracy and performance.

### Improvements to `to_datetime` function:
* Improved `pd.to_datetime` to handle all local input cases. (`CHANGELOG.md`)
* Updated `to_datetime` to handle non-Modin objects using `pandas.to_datetime` and to return appropriate Modin objects. (`src/snowflake/snowpark/modin/pandas/general.py`) [[1]](diffhunk://#diff-b1393ad1ad080e6edda4cdf185672f7dc989abdf6ad5a5bdc5c6dfa303103105L1787-R1805) [[2]](diffhunk://#diff-b1393ad1ad080e6edda4cdf185672f7dc989abdf6ad5a5bdc5c6dfa303103105L1798-R1816)
* Refined the handling of timezone-naive and timezone-aware inputs in `to_datetime`. (`src/snowflake/snowpark/modin/pandas/general.py`) [[1]](diffhunk://#diff-b1393ad1ad080e6edda4cdf185672f7dc989abdf6ad5a5bdc5c6dfa303103105L1745-R1751) [[2]](diffhunk://#diff-b1393ad1ad080e6edda4cdf185672f7dc989abdf6ad5a5bdc5c6dfa303103105L1766-R1773)

### Updates to internal utilities:
* Introduced `AUTO_FORMAT_WARNING_MSG` for consistent warning messages about Snowflake's automatic format detection. (`src/snowflake/snowpark/modin/plugin/_internal/timestamp_utils.py`)
* Updated warning messages to reflect potential mismatches with pandas more clearly. (`src/snowflake/snowpark/modin/plugin/utils/warning_message.py`)

### Adjustments to test cases:
* Modified test cases to use `pd.Index` for datetime inputs to ensure consistency and accuracy in results. (`tests/integ/modin/tools/test_to_datetime.py`) [[1]](diffhunk://#diff-f3ba9d6671bbb8f8ada5e8ef9964c4509c0a999b5f5c75889b375115c4959b98L107-R107) [[2]](diffhunk://#diff-f3ba9d6671bbb8f8ada5e8ef9964c4509c0a999b5f5c75889b375115c4959b98L123-R123) [[3]](diffhunk://#diff-f3ba9d6671bbb8f8ada5e8ef9964c4509c0a999b5f5c75889b375115c4959b98L138-R138)
* Adjusted SQL query count expectations in tests to reflect the improved performance of `to_datetime`. (`tests/integ/modin/tools/test_to_datetime.py`) [[1]](diffhunk://#diff-f3ba9d6671bbb8f8ada5e8ef9964c4509c0a999b5f5c75889b375115c4959b98L331-R331) [[2]](diffhunk://#diff-f3ba9d6671bbb8f8ada5e8ef9964c4509c0a999b5f5c75889b375115c4959b98L387-R389) [[3]](diffhunk://#diff-f3ba9d6671bbb8f8ada5e8ef9964c4509c0a999b5f5c75889b375115c4959b98L410-R414)

These changes enhance the robustness and performance of datetime handling in the Snowpark Modin integration, ensuring better alignment with pandas behavior and more accurate test coverage.

### CODE REVIEW

File: src/snowflake/snowpark/modin/pandas/general.py [hunk 1]
```diff
@@ -1784,8 +1781,28 @@ def to_datetime(
     # TODO: SNOW-1063345: Modin upgrade - modin.pandas functions in general.py
     raise_if_native_pandas_objects(arg)
 
-    if arg is None:
-        return None  # same as pandas
+    if not isinstance(arg, (DataFrame, Series, pd.Index)):
```
Comment (sfc-gh-azhan) [inline src/snowflake/snowpark/modin/pandas/general.py:1784]: match with what we do today for DatetimeIndex and timedelta.

File: src/snowflake/snowpark/modin/plugin/utils/warning_message.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ def ignored_argument(cls, operation: str, argument: str, message: str) -> None:
     @classmethod
     def mismatch_with_pandas(cls, operation: str, message: str) -> None:
         cls.single_warning(
-            f"`{operation}` implementation has mismatches with pandas:\n{message}."
+            f"`{operation}` implementation may has mismatches with pandas:\n{message}."
```
Comment (sfc-gh-nkrishna) [inline src/snowflake/snowpark/modin/plugin/utils/warning_message.py:54]: Nit:
```suggestion
            f"`{operation}` implementation may have mismatches with pandas:\n{message}."
```

**Review by sfc-gh-nkrishna — APPROVED** [review]
One minor nit, but LGTM!

**Review by sfc-gh-mvashishtha — APPROVED** [review]

File: tests/notebooks/modin/TimeSeriesTesting.ipynb [hunk 3]
```diff
@@ -142,66 +163,132 @@
...
+      "\u001b[0;31mNotImplementedError\u001b[0m: Snowpark pandas does not yet support the method DatetimeIndex.tz_localize"
+     ]
+    }
+   ],
    "source": [
     "# TODO SNOW-1635620: uncomment when TimeDelta is implemented\n",
```
[the remark this answers is not in the document]
Comment (sfc-gh-azhan) [reply tests/notebooks/modin/TimeSeriesTesting.ipynb]: good catch

**Review by sfc-gh-mvashishtha — APPROVED** [review]

**Comment by sfc-gh-helmeleegy** [comment]
Are we sure we don't want to break up this PR?

**Comment by sfc-gh-azhan** [comment]
> Are we sure we don't want to break up this PR?

It's only two changes on src:
1. use pandas to_datetime for local inputs
2. warning message change

Others are just test updated based on the above two.

**Comment by sfc-gh-helmeleegy** [comment]
> > Are we sure we don't want to break up this PR?
> 
> It's only two changes on src:
> 
> 1. use pandas to_datetime for local inputs
> 2. warning message change
> 
> Others are just test updated based on the above two.
> 
> I did include most of tests from the pandas time series doc in our notebook in this PR. But should be fine if something can be improved there, we can use followup PRs

This makes sense.

**Pull request merged**